### PR TITLE
tests: update nested tool - part1

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The change includes:
 1. Many functions added to tests.nested tool
        - is-nested
        - get
    	- boot-id
        - secboot-sign
 2. Restructured the tests.nested tool command to simplify the cli
        - exec
        - vm (before was start-vm, stop-vm, and remove-vm)

The tests were updated to use the tests.nested tool instead of the nested.sh helper.
The next step is to reorganize the helper in 2 parts.

### CODE REVIEW

**Review by mardy — APPROVED** [review]
LGTM!

File: tests/nested/manual/uc20-fde-hooks/task.yaml [hunk 1]
```diff
@@ -14,10 +14,6 @@ prepare: |
   echo "Build a kernel snap with the fde-setup hook"
   # shellcheck source=tests/lib/prepare.sh
   . "$TESTSLIB/prepare.sh"
-  # shellcheck source=tests/lib/nested.sh
-  . "$TESTSLIB/nested.sh"
-
-  mkdir "$(nested_get_extra_snaps_path)"
```
Comment (stolowski) [inline tests/nested/manual/uc20-fde-hooks/task.yaml:20]: Is this removal intended?

File: tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml [hunk 2]
```diff
@@ -145,14 +142,14 @@ execute: |
...
-  nested_shutdown
+  boot_id="$(tests.nested boot-id)"
+  tests.nested vm stop
 
   # replace the seed.iso with the new attacker iso
   mv seed2.iso seed.iso
 
   echo "Restarting nested VM with attacker cloud-init CD-ROM drive"
-  nested_force_start_vm
+  tests.nested vm start
```
Comment (stolowski) [inline tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml:152]: AFAIU start is really using "force_start" underneath, so this is fine.

**Review by stolowski — APPROVED** [review]
Looks very nice, thank you. One question and one suggestion (maybe for later).

File: tests/nested/manual/uc20-fde-hooks/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/nested/manual/uc20-fde-hooks/task.yaml:20]: hi, thanks for the review, yes it is intentional, the path is not being used so we don't need to create it.

File: tests/main/preseed-reset/task.yaml [hunk 3]
```diff
@@ -12,9 +12,7 @@ environment:
 prepare: |
   # the get_image_url_for_nested_vm is a convenient helper that returns
   # a cloud image url matching current $SPREAD_SYSTEM.
-  #shellcheck source=tests/lib/nested.sh
-  . "$TESTSLIB/nested.sh"
-  wget --quiet "$(nested_get_image_url_for_vm)" -O cloudimg.img
+  wget --quiet "$(tests.nested get image-url)" -O cloudimg.img
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/main/preseed-reset/task.yaml:15]: Nice idea, I though to also include the snakeoil certificates in the download. 
I need to push the second part of the nested change, I'll add the download command in that change.

File: tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml:152]: yes, we should always use tests.nested vm start, it makes sure the start is done and the control to the test is back once the VM is ready.
Thanks for the review
